doc/upgrade-to-2.x.md

```diff
@@ -0,0 +1,474 @@
+
+## 💥 Tracing SDK API changes
+
+This section describes API changes in the set of packages that implement the tracing SDK: `@opentelemetry/sdk-trace-base`, `@opentelemetry/sdk-trace-node`, `@opentelemetry/sdk-trace-web`.
+
+Tracing was the first signal to have an SDK. Over time, and as the Metrics and Logs SDKs were added, the API design separating functionality between the `{Tracer,Meter,Logs}Provider`s APIs and the higher level `NodeSDK` (in `@opentelemetry/sdk-node`) improved. The Tracing SDK was left with some cruft that is being removed in JS SDK 2.0. (See [#5290](https://github.com/open-telemetry/opentelemetry-js/issues/5290) for motivation.)
+
+- removed `BasicTracerProvider#addSpanProcessor(...)` -> use constructor options to the TracerProvider class
+- made `BasicTracerProvider#getActiveSpanProcessor()` private
+- made `BasicTracerProvider#resource` private
+- `BasicTracerProvider` and `NodeTracerProvider` will no longer use the `OTEL_TRACES_EXPORTER` envvar to create exporters -> This functionality already resides in `NodeSDK` (from `@opentelemetry/sdk-node`).
+- `BasicTracerProvider` and `NodeTracerProvider` will no longer use the `OTEL_PROPAGATORS` envvar to create propagators -> This functionality already resides in `NodeSDK` (from `@opentelemetry/sdk-node`).
+- The following internal properties were removed from `BasicTracerProvider`: `_registeredExporters`, `_getSpanExporter`, `_buildExporterFromEnv`
+- The following exports were dropped from `@opentelemetry/sdk-trace-*`:
+  - `EXPORTER_FACTORY` is not used anymore and has been removed
+  - `PROPAGATOR_FACTORY` is not used anymore and has been removed
+  - `ForceFlushState` was intended for internal use and has been removed
+  - the `Tracer` class was unintentionally exported and has been removed
+    - to obtain a `Tracer`, please use `BasicTracerProvider#getTracer()`, `NodeTracerProvider#getTracer()` or `WebTracerProvider#getTracer()`
+    - to reference the `Tracer` type, please use the `Tracer` type from `@opentelemetry/api`
+- removed `BasicTracerProvider#register()` -> use `NodeTracerProvider#register()` or `WebTracerProvider#register()`, or call `trace.setGlobalTracerProvider()` et al manually (see [#5503](https://github.com/open-telemetry/opentelemetry-js/pull/5503))
+
+<br/>
+
+The export of the `Span` class has been removed. It was not intended to be used directly. One should always use methods on a `Tracer` instance (e.g. `startSpan()`) for creating spans.
+
+- `new Span(...)` -> use `tracer.startSpan(...)`
+
+<br/>
+
+The `Span` `parentSpanId` field was replaced by `parentSpanContext`, to adhere to the OTel spec. [#5450](https://github.com/open-telemetry/opentelemetry-js/pull/5450)
+
+- `span.parentSpanId` -> `span.parentSpanContext?.spanId`
+
+<br/>
+
+As mentioned above in the "core" section, `InstrumentationLibrary` has been changed to `InstrumentationScope`:
+
+- `Tracer.instrumentationLibrary` -> `Tracer.instrumentationScope`
```

Comment:
```suggestion
```

Tracer is now internal, so I don't think we need to call that out specifically 🙂